game_components/rooms/basic_room.py

```diff
@@ -0,0 +1,82 @@
+from abc import ABC, abstractmethod  # abstract classes
+import hashlib
+from game_components.entitiies.player import Player
+from game_components.entitiies.mob import Mob
+
+
+class BaseRoom(ABC):
+    __title = None
```

Comment:
Initializing variables this way isn't needed afaik. They'll be correctly added by `__init__` without the need to set them to `None` beforehand.
For "private" variables only one underscore should be used before the variable name. Multiple underscores are generally meant for name mangling and it can lead to some unexpected problems iirc.

Reply:
no. _ means something that is gonna get used as a temp or passed as an arg that i dont wanna deal with dealing with ghosting. 2 underscores is private
https://stackoverflow.com/questions/1641219/does-python-have-private-variables-in-classes

Reply:
From the PEP 8 guide:
`Even with __all__ set appropriately, internal interfaces (packages, modules, classes, functions, attributes or other names) should still be prefixed with a single leading underscore.`
As the so response you linked suggest, double underscores produce mame mangling which can have unintended effects. Single underscore are used to indicate an attribute should be treated as "private" but don't actually cause any effect (just a convention).

Reply:
but __ is actually private.


```
class Foo:
    def __init__(self):
        print("Foo")
        self.value = 0
        self.__secret_value = 7

    def get_secret(self):
        return self.__secret_value

if __name__ =="__main__":
    f = Foo()
    print(f.value)
    print(f.get_secret())
    print(f.__secret_value)
```

output:
```
Foo
0
7
Traceback (most recent call last):
  File "C:\Users\azbai\PycharmProjects\fleshman\scratch.py", line 14, in <module>
    print(f.__secret_value)
AttributeError: 'Foo' object has no attribute '__secret_value'. Did you mean: '_Foo__secret_value'?
```

Reply:
They aren't really "private". Two underscores are used for name mangling, and avoiding name conflicts in subclasses. If you want private variables, just use a single underscore. It isn't enforced by the interpreter, but generally programmer's will know to leave a variable with one underscore alone. If they don't, that's on them. Here's the relevant pep8 https://peps.python.org/pep-0008/#method-names-and-instance-variables